lib/src/chrome.dart

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:coverage/src/util.dart';
+import 'package:source_maps/parser.dart';
+
+/// Returns a Dart based hit-map containing coverage report for the provided
+/// Chrome [preciseCoverage].
+///
+/// [sourceProvider] returns the source content for the Chrome scriptId, or null
+/// if not available.
+///
+/// [sourceMapProvider] returns the associated source map content for the Chrome
+/// scriptId, or null if not available.
+///
+/// [sourceUriProvider] returns the uri for the provided sourceUrl and
+/// associated scriptId.
+Future<Map<String, dynamic>> parseChromeCoverage(
+  List<Map<String, dynamic>> preciseCoverage,
+  Future<String> Function(String scriptId) sourceProvider,
+  Future<String> Function(String scriptId) sourceMapProvider,
+  Future<Uri> Function(String sourceUrl, String scriptId) sourceUriProvider,
+) async {
+  final coverageReport = <Uri, Map<int, bool>>{};
+  for (Map<String, dynamic> entry in preciseCoverage) {
+    final String scriptId = entry['scriptId'];
+
+    final mapResponse = await sourceMapProvider(scriptId);
+    if (mapResponse == null) continue;
+
+    final SingleMapping mapping = parse(mapResponse);
+
+    final compiledSource = await sourceProvider(scriptId);
+    if (compiledSource == null) continue;
+
+    final coverageInfo = _coverageInfoFor(entry);
+    final offsetCoverage = _offsetCoverage(coverageInfo, compiledSource.length);
+    final coveredPositions = _coveredPositions(compiledSource, offsetCoverage);
+
+    for (var lineEntry in mapping.lines) {
+      for (var columnEntry in lineEntry.entries) {
+        if (columnEntry.sourceUrlId == null) continue;
+        final sourceUrl = mapping.urls[columnEntry.sourceUrlId];
+
+        // Ignore coverage information for the SDK.
+        if (sourceUrl.startsWith('org-dartlang-sdk:')) continue;
+
+        final uri = await sourceUriProvider(sourceUrl, scriptId);
+        final coverage = coverageReport.putIfAbsent(uri, () => <int, bool>{});
+
+        coverage[columnEntry.sourceLine + 1] = coveredPositions
+            .contains(_Position(lineEntry.line + 1, columnEntry.column + 1));
+      }
+    }
+  }
+
+  final coverageHitMaps = <Uri, Map<int, int>>{};
+  coverageReport.forEach((uri, coverage) {
+    final hitMap = <int, int>{};
+    for (var line in coverage.keys.toList()..sort()) {
+      hitMap[line] = coverage[line] ? 1 : 0;
+    }
+    coverageHitMaps[uri] = hitMap;
+  });
+
+  final allCoverage = <Map<String, dynamic>>[];
+  coverageHitMaps.forEach((uri, hitMap) {
+    allCoverage.add(toScriptCoverageJson(uri, hitMap));
+  });
```

Comment:
[optional]

```dart
final allCoverage = [
  for(var hitMap in coverageHitMaps.entries)
    toScriptCoverageJson(hitMap.key, hitMap.value),
];
```

Reply:
The SDK constraint is wide on this package so I don't believe we can use this feature yet.